osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -144,100 +140,104 @@ public void ShowSpotlights()
             Show();
         }
 
-        private void loadNewContent()
+        private void selectDisplayToLoad()
         {
-            loading.Show();
-
             cancellationToken?.Cancel();
-            lastRequest?.Cancel();
+            spotlightsRequest?.Cancel();
 
-            if (Scope.Value == RankingsScope.Spotlights)
+            loading.Show();
+
+            if (!API.IsLoggedIn)
             {
-                loadContent(new SpotlightsLayout
-                {
-                    Ruleset = { BindTarget = ruleset }
-                });
+                loadDisplayAsync(Empty());
                 return;
             }
 
-            var request = createScopedRequest();
-            lastRequest = request;
+            var display = selectDisplay();
 
-            if (request == null)
+            if (display is SpotlightsRankingsDisplay)
             {
-                loadContent(null);
+                loadSpotlightsDisplay(display);
                 return;
             }
```

Comment:
this is weird. `selectDisplay` is a generic factory method, but then whatever gets created gets pattern-matched anyway (as spotlights have different logic). that logic should arguably reside in the displays themselves.